File: frontend/src/backend/productManagementService.js
import {config, appFetch} from './appFetch';

export const addProduct = (companyId, name, description, price, file, productCategoryId, 
    onSuccess, onErrors) =>{

        let path = `/management/products`;

        const formData = new FormData();
        formData.append('file', file);
        formData.append('data', new Blob([JSON.stringify({ name, description, price, companyId, productCategoryId})], {type: "application/json"} ));

        appFetch(path, config('POST', formData),
            onSuccess, onErrors);

    }

export const editProduct = (productId, companyId, name, description, price, file, productCategoryId, onSuccess, onErrors) =>{

    let path = `/management/products/${productId}`;

    const formData = new FormData();
        formData.append('file', file);
        formData.append('data', new Blob([JSON.stringify({ name, description, price, companyId, productCategoryId})], {type: "application/json"} ));

    appFetch(path, config('PUT', formData),
        onSuccess, onErrors);
}

export const removeProduct = (productId, companyId, onSuccess, onErrors) =>
    appFetch(`/management/products/${productId}?companyId=${companyId}`, 
        config('DELETE'), onSuccess, onErrors);

export const blockProduct = (productId, companyId, onSuccess) =>
    appFetch(`/management/products/${productId}/block`, 
        config('POST', {companyId}), onSuccess);

export const unlockProduct = (productId, companyId, onSuccess) =>
    appFetch(`/management/products/${productId}/unlock`, 
        config('POST', {companyId}), onSuccess);

export const findProduct = (productId, onSuccess) => 
    appFetch(`/management/products/${productId}`, 
        config('GET'), onSuccess);

export const findAllProductCategories = (onSuccess) => 
    appFetch('/management/products/categories', 
        config('GET'), onSuccess);

export const findAllCompanyProducts = (companyId, onSuccess) => 
    appFetch(`/management/products?companyId=${companyId}`, 
        config('GET'), onSuccess);

        export const addProductCategory = (name, onSuccess, onErrors) =>
    appFetch(`/management/productCategories`,
        config('POST', {name}), onSuccess, onErrors);

export const modifyProductCategory = (productCategory, onSuccess, onErrors) =>
    appFetch(`/management/productCategories/${productCategory.id}`, config('PUT', productCategory),
        onSuccess, onErrors);

    

File: backend/src/main/java/es/udc/tfgproject/backend/model/services/ProductManagementService.java
package es.udc.tfgproject.backend.model.services;

import java.math.BigDecimal;
import java.util.List;

import es.udc.tfgproject.backend.model.entities.Product;
import es.udc.tfgproject.backend.model.entities.ProductCategory;
import es.udc.tfgproject.backend.model.exceptions.InstanceNotFoundException;
import es.udc.tfgproject.backend.model.exceptions.PermissionException;

public interface ProductManagementService {

	Product addProduct(Long userId, Long companyId, String name, String description, BigDecimal price, String path,
			Long productCategoryId) throws InstanceNotFoundException, PermissionException;

	Product editProduct(Long userId, Long companyId, Long productId, String name, String description, BigDecimal price,
			String newPath, Long productCategoryId) throws InstanceNotFoundException, PermissionException;

	void removeProduct(Long userId, Long companyId, Long productId)
			throws InstanceNotFoundException, PermissionException;

	Product blockProduct(Long userId, Long companyId, Long productId)
			throws InstanceNotFoundException, PermissionException;

	Product unlockProduct(Long userId, Long companyId, Long productId)
			throws InstanceNotFoundException, PermissionException;

	Product findProduct(Long userId, Long productId) throws InstanceNotFoundException, PermissionException;

	List<ProductCategory> findAllProductCategories();

	ProductCategory addProductCategory(Long userId, String name) throws InstanceNotFoundException, PermissionException;

	ProductCategory modifyProductCategory(Long userId, Long productCategoryId, String name)
			throws InstanceNotFoundException, PermissionException;

	List<Product> findAllCompanyProducts(Long userId, Long companyId)
			throws InstanceNotFoundException, PermissionException;

}
